editor/editor_node.cpp

```diff
@@ -7591,36 +7590,50 @@ EditorNode::EditorNode() {
 		title_bar->add_child(right_menu_spacer);
 	}
 
-	String current_renderer = GLOBAL_GET("rendering/renderer/rendering_method");
-
-	PackedStringArray renderers = ProjectSettings::get_singleton()->get_custom_property_info().get(StringName("rendering/renderer/rendering_method")).hint_string.split(",", false);
-
-	// As we are doing string comparisons, keep in standard case to prevent problems with capitals
-	// "vulkan" in particular uses lowercase "v" in the code, and uppercase in the UI.
-	current_renderer = current_renderer.to_lower();
-
-	for (int i = 0; i < renderers.size(); i++) {
-		String rendering_method = renderers[i];
+	String current_renderer_ps = GLOBAL_GET("rendering/renderer/rendering_method");
+	current_renderer_ps = current_renderer_ps.to_lower();
+	String current_renderer_os = OS::get_singleton()->get_current_rendering_method().to_lower();
 
-		// Add the renderers name to the UI.
-		if (rendering_method == "forward_plus") {
-			renderer->add_item(TTR("Forward+"));
+	// Add the renderers name to the UI.
+	auto _add_renderer_entry = [this](const String &p_renderer_name, bool p_mark_overridden) {
```

Comment:
Is a lambda really necessary here? AFAIK our policy is to only allow them where necessary, or where they are so beneficial that it makes sense. This case doesn't look like it is necessary or adds anything over using a regular function. 

CC @RandomShaper since it was his suggestion

Reply:
Necessary I wouldn't say. Beneficial I would. I have already used lambdas in my PRs in this fashion, where they avoid code duplication without adding the need for a new member function.

Reply:
Okay, our docs are less strict than what I thought:

> Lambdas should be used conservatively when they make code effectively faster or simpler, and do not impede readability. Please ask before using lambdas in a pull request.

(https://docs.godotengine.org/en/latest/contributing/development/cpp_usage_guidelines.html#lambdas)

So I'd agree its fine to use here since you support doing so. 

My personal opinion would be to never use lambda's unless necessary. But don't let me stand in the way of progress :stuck_out_tongue:

Reply:
Not too fond of that lambda either, for me the policy means that they should only be used where they are absolutely a net positive (i.e. they enable doing something that a regular member function wouldn't).

Especially here as it's impacting a member variable (`renderer`) and not self-contained, I think this would be preferable as a member function.

Reply:
I hope we can discuss guidelines on lambdas at some point. As I see it, the benefit is great in cases like this: no need to touch the header (less clutter, faster iteration) and better readability because of avoiding out-of-context code ("Why does this class need that function?"). For the same reason, we could forbid using other C++ features, as there's always another way, and trim it down to C (even without classes!). I'm leaving these thoughts here so they install and start doing their job in your subconscious minds. 😃 For now, I understand if the current rule explicitly favors conservatism, my suggestion would be breaking it and this is not the right place to re-discuss it.